style_guide.md

```diff
@@ -0,0 +1,99 @@
+# Keras Style Guide
+
+This style guide contains information needed to write code in the style of the Keras
+codebase.
+
+### Layer-ish Functions
+
+When writing Keras code it is common to create functions that apply a logical block of
+deep learning code, such as a ResNet block.  We write our blocks using a closure pattern
+that makes the behavior of these blocks mirror Keras layers.
+
+Bad:
```

Comment:
"bad" seems too strong, the below code is fine. We can just show the closure and say "recommended"